### PR TITLE
add polyfill to avoid some IE errors

### PR DESCRIPTION
**Pull request by kaidohallik**
related to #5186 and #5276

### CODE REVIEW

**Comment by kaidohallik** [comment]
Alternative solution to `core-js/client/shim` is to use es6-shim, named by @deepu105 in https://github.com/jhipster/generator-jhipster/issues/5186#issuecomment-288990778
1. add dev dependency to package.json:
```
	"es6-shim": "0.35.3",
```
2. add this line to polyfills.ts:
```
import 'es6-shim/es6-shim';
```

**Comment by kaidohallik** [comment]
page with core-js/client/shim is 600 kB bigger  
page with es6-shim/es6-shim is 350 kB bigger

**Comment by kaidohallik** [comment]
I made shim conditional to IE 11 only with javascript hack in index.html. PR updated.

File: generators/client/templates/angular/src/main/webapp/_index.html [hunk 1]
```diff
@@ -1,6 +1,15 @@
 <!doctype html>
 <html class="no-js">
 <head>
+    <script>
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/_index.html]: can we move this to right after the `<jhi-main></jhi-main>` in this file please.

Btw did this work in IE?
@holgerdunst does this solve your issue?

**Comment by kaidohallik** [comment]
Script moved after the `<jhi-main></jhi-main>`. PR updated.  
This solves IE 11 error.  
I didn't test this with IE 10, for IE 10 this needs additional changes, for example `userAgent` string is different in IE 10.

**Comment by sendilkumarn** [comment]
@kaidohallik `Trident` does include IE10 right ?

Can we have more specific IE11 check in that case as in [this](http://stackoverflow.com/a/17447718)

**Comment by kaidohallik** [comment]
@sendilkumarn My previous comment about IE 10 can be incorrect. It's possible that this fix works also in IE 10. `Trident/` is also part of IE 10 user agent string: https://msdn.microsoft.com/en-us/library/ms537503(v=vs.85).aspx

**Comment by kaidohallik** [comment]
@sendilkumarn Now I think that this shim can be applied to all IE versions, so no need to detect IE 11.  
Feature detection is possible, but then logic goes maybe too complex. Question is: what feature exactly to detect...

**Pull request merged**